### PR TITLE
Same two way key

### PR DESCRIPTION
**Pull request by fogelito**
Fix for relations attributes when twoWayKey variable is not passed in API,
We do not want the worker to throw the Error
Until now, it was checked only for twoWay = true

Added a validation to prevent the creation of 2 manyToMany relations between same collections

Fixes: https://github.com/appwrite/appwrite/issues/6281

### CODE REVIEW

File: app/controllers/api/databases.php [hunk 1]
```diff
@@ -1615,8 +1615,49 @@ function updateAttribute(
...
+
+            if (
+                $attribute->getAttribute('type') === Database::VAR_RELATIONSHIP &&
+                \strtolower($attribute->getAttribute('options')['twoWayKey']) === \strtolower($twoWayKey) &&
+                $attribute->getAttribute('options')['relatedCollection'] === $relatedCollection->getId()
+            ) {
+                if ($twoWayKeyNull) {
+                    $twoWayKey .= '_' . uniqid();
+                } else {
+                    throw new Exception(Exception::DOCUMENT_ALREADY_EXISTS);
```
Comment (abnegate) [inline app/controllers/api/databases.php:1656]: Same here for the exception type

File: app/controllers/api/databases.php [hunk 2]
```diff
@@ -1615,8 +1615,49 @@ function updateAttribute(
...
+
+        if ($relatedCollection->isEmpty()) {
+            throw new Exception(Exception::COLLECTION_NOT_FOUND);
+        }
+
+        $attributes = $collection->getAttribute('attributes', []);
+        /** @var Document[] $attributes */
+        foreach ($attributes as $attribute) {
+            if (\strtolower($attribute->getId()) === \strtolower($key)) {
+                throw new Exception(Exception::DOCUMENT_ALREADY_EXISTS);
```
[the remark this answers is not in the document]
Comment (fogelito) [reply app/controllers/api/databases.php]: Yep done

File: app/controllers/api/databases.php (hunk 1 above)
Comment (fogelito) [reply app/controllers/api/databases.php]: Done
I have removed the change of the attribute naming, as we spoke, currently will just throw an Error.

File: app/controllers/api/databases.php [hunk 3]
```diff
@@ -1615,8 +1615,55 @@ function updateAttribute(
...
+                throw new Exception(Exception::ATTRIBUTE_ALREADY_EXISTS);
+            }
+
+            if (
+                $type === Database::RELATION_MANY_TO_MANY &&
+                $attribute->getAttribute('type') === Database::VAR_RELATIONSHIP &&
+                $attribute->getAttribute('options')['relationType'] === Database::RELATION_MANY_TO_MANY &&
+                $attribute->getAttribute('options')['relatedCollection'] === $relatedCollection->getId()
+            ) {
+                throw new Exception('Cannot create more than one ManyToMany relation.');
```
Comment (fogelito) [inline app/controllers/api/databases.php:1663]: @abnegate what kind of Exception should we throw?

File: app/controllers/api/databases.php [hunk 4]
```diff
@@ -1615,8 +1615,55 @@ function updateAttribute(
         Event $events
     ) {
         $key ??= $relatedCollectionId;
+        $twoWayKeyNull = is_null($twoWayKey);
         $twoWayKey ??= $collectionId;
 
+        $databaseDocument = Authorization::skip(fn() => $dbForProject->getDocument('databases', $databaseId));
```
Comment (abnegate) [inline app/controllers/api/databases.php:1621]: Can we do this in the `createAttribute` function, I think we have a lot of this code there already

File: app/controllers/api/databases.php (hunk 3 above)
Comment (abnegate) [reply app/controllers/api/databases.php:1663]: Maybe still `ATTRIBUTE_ALREADY_EXISTS` but with this message `Creating more than one "manyToMany" relationship on the same collection is currently not permitted.`

File: phpunit.xml [hunk 5]
```diff
@@ -6,7 +6,7 @@
          convertNoticesToExceptions="true"
          convertWarningsToExceptions="true"
          processIsolation="false"
-         stopOnFailure="false"
+         stopOnFailure="true"
```
Comment (abnegate) [inline phpunit.xml:9]: Let's revert this

**Comment by fogelito** [comment]
PR moved 
https://github.com/appwrite/appwrite/pull/7153

**Pull request closed without merging**